src/main/java/com/metamx/common/parsers/RegexParser.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2011,2012 Metamarkets Group Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.metamx.common.parsers;
+
+import com.google.common.base.Function;
+import com.google.common.base.Optional;
+import com.google.common.base.Splitter;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Lists;
+import com.metamx.common.collect.Utils;
+import com.metamx.common.logger.Logger;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ */
+public class RegexParser implements Parser<String, Object>
+{
+  private static final Logger log = new Logger(RegexParser.class);
+
+  private final String pattern;
+  private final String listDelimiter;
+  private final Splitter listSplitter;
+  private final Function<String, Object> valueFunction;
+
+  private ArrayList<String> fieldNames = null;
+
+  public RegexParser(
+      final String pattern,
+      final Optional<String> listDelimiter
+  )
+  {
+    this.pattern = pattern;
+    this.listDelimiter = listDelimiter.isPresent() ? listDelimiter.get() : Parsers.DEFAULT_LIST_DELIMITER;
+    this.listSplitter = Splitter.on(this.listDelimiter);
+    this.valueFunction = new Function<String, Object>()
+    {
+      @Override
+      public Object apply(String input)
+      {
+        if (input.contains(RegexParser.this.listDelimiter)) {
+          return Lists.newArrayList(
+              Iterables.transform(
+                  listSplitter.split(input),
+                  ParserUtils.nullEmptyStringFunction
+              )
+          );
+        } else {
+          return ParserUtils.nullEmptyStringFunction.apply(input);
+        }
+      }
+    };
+  }
+
+  public RegexParser(
+      final String pattern,
+      final Optional<String> listDelimiter,
+      final Iterable<String> fieldNames
+  )
+  {
+    this(pattern, listDelimiter);
+
+    setFieldNames(fieldNames);
+  }
+
+  @Override
+  public Map<String, Object> parse(String input)
+  {
+    try {
+      Pattern compiled = Pattern.compile(pattern);
+      Matcher matcher = compiled.matcher(input);
+
+      if (!matcher.matches()) {
+        throw new ParseException("Incorrect Regex: %s . No match found.", pattern);
+      }
+
+      List<String> values = Lists.newArrayList();
+      for (int i = 0; i < matcher.groupCount(); i++) {
+        values.add(matcher.group(i + 1));
+      }
```

Comment:
personally speaking :) , I would have done `(int i = 1; i <= matcher.groupCount(); i++)`
so that matcher.group(i) would be fine, easier to read that way.